Sources/VitalHealthKit/HealthKit/HealthKitReads.swift

```diff
@@ -1344,9 +1345,19 @@ private func filterForWatch(samples: [HKSample]) -> [HKSample] {
 }
 
 private func filter(samples: [HKSample], by dataSources: [DataSource]) -> [HKSample] {
+  
   return samples.filter { sample in
     let identifier = sample.sourceRevision.source.bundleIdentifier
-    
+
+    guard 
+      let wasUserEntered = sample.metadata?[HKMetadataKeyWasUserEntered] as? Bool,
+      wasUserEntered == false
+    else {
+      /// If it's manually entered, allow the sample
```

Comment:
You should use if let here. Otherwise everything that doesn’t have HKMetadataKeyWasUserEntered would be included unconditionally, which is… generally everything.